WordPress/gutenberg
# DataViews: add details for pages in list layout

Part of https://github.com/WordPress/gutenberg/issues/55083
Superseded by https://github.com/WordPress/gutenberg/pull/57578

## What?

Implement the page details on the list view.

https://github.com/WordPress/gutenberg/assets/583546/979fa1ab-3fcc-4cd7-bafd-f57ff9814594


## Why?

It's part of the design spec:

![240103-page-list-page-detail](https://github.com/WordPress/gutenberg/assets/583546/c549c5b0-6f2d-4425-b385-ebb8cf93e3ad)

## How?

This PR introduces the details as a consumer action. They render the details, like they handle the actions and the preview.

- DataViews API gains a `onDetailsChange` callback, that signals the user triggered the details for an item.
- The `Page` component is updated to take an `onClose` callback. If present, it renders a "left chevron" button, otherwise, it's not present. 

## Alternatives

I considered making DataViews handling the details screen. This would require it to absorb parts of the site editor UI that are now outside the library, so I decided against it:

<img width="1202" alt="Captura de ecrã 2024-01-04, às 12 41 06" src="https://github.com/WordPress/gutenberg/assets/583546/8c842aae-5d65-4026-9bde-2e25cabfb095">

## Testing Instructions

Test pages show the details:

- Enable the "new admin view" experiment and visit "Manage all pages".
- Switch to the list layout. Note that it has a new "right chevron" button. Interact with it as shown in the video above.

<img width="1510" alt="Captura de ecrã 2024-01-04, às 12 24 24" src="https://github.com/WordPress/gutenberg/assets/583546/fb542c8a-1919-4b83-9cbc-83fb2fdb0dc1">

Test the templates do **not** have the details button:

- Enable the "new admin view" experiment and visit "Manage all templates".
- Switch to the list layout. This is how it should look like (no details button):

<img width="1510" alt="Captura de ecrã 2024-01-04, às 12 23 11" src="https://github.com/WordPress/gutenberg/assets/583546/e64e8257-5705-4d61-b5a2-3eb2fa040fe7">

## Follow-ups

- Implement details for templates.
- Add actions.
